docs/pattern-features.md

```diff
@@ -227,6 +228,29 @@ pattern: $X = 1 + 2 + ...
 foo = 1 + 2 + 3 + 4
 ```
 
+### Deep Expression Operator
+
+You may want to match an expression that could be nested deep within another expression. An example of this is looking for a pattern anywhere within an if statement. To do this, use the deep expression operator: `<... [your_pattern] ...>`. This will match your pattern in the current expression context and recursively in any subexpressions.
+
+For example, this pattern:
+
+```yaml
+if <... $USER.is_admin() ...>:
+  ...
+```
+
+will match:
+
+```python
+if user.authenticated() and user.is_admin() and user.has_group(gid):
+  ...
+```
+
+The deep expression operator will work in:
+* if statements -- `if <... $X ...>:`
+* nested calls -- `sql.query(<... $X ...>)`
+* operands of a binary expression -- `"..." + <... $X ...>`
+
```

Comment:
or more generally in any expression context.

Reply:
Good call! Maybe we can provide some more expression contexts where the deep expression operator works, then leave a bullet saying something like "And any other expression context."

My concern is that your average engineer won't understand what "expression context" means. When I think about docs I try to target my language to something an average, entry-level engineer can understand. I've been thinking about `semgrep` as "Static analysis, for everyone" and we can target that with simplicity, open source, good docs, and self-service as some of our competitive advantages :)